mountpoint-s3/tests/fuse_tests/read_test.rs

```diff
@@ -309,3 +309,41 @@ fn read_errors_test_s3() {
 fn read_errors_test_mock(prefix: &str) {
     read_errors_test(fuse::mock_session::new, prefix);
 }
+
+fn read_after_flush_test<F>(creator_fn: F)
+where
+    F: FnOnce(&str, TestSessionConfig) -> (TempDir, BackgroundSession, TestClientBox),
+{
+    const KEY: &str = "data.bin";
+    let (mount_point, _session, mut test_client) = creator_fn("read_after_flush_test", Default::default());
+
+    let mut rng = ChaChaRng::seed_from_u64(0x87654321);
+    let mut two_mib_body = vec![0; 2 * 1024 * 1024];
+    rng.fill_bytes(&mut two_mib_body);
+    test_client.put_object(KEY, &two_mib_body).unwrap();
+
+    let path = mount_point.path().join(KEY);
+    let mut f = open_for_read(path, true).unwrap();
+
+    let mut content = vec![0; 128];
+    f.read_exact(&mut content).unwrap();
+
+    let mut f_dup = f.try_clone().unwrap();
+
+    drop(f);
```

Comment:
A small comment mentioning we're intentionally closing the file and triggering a flush here would be useful